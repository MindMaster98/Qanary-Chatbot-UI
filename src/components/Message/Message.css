.messageContainer {
  width: 100%;
  height: fit-content;
  float: right;
  display: flex;
  flex-direction: row-reverse;
  margin-bottom: 3vh;
}

.messageContainer.reply {
    flex-direction: row;
}

.messageDataContainer {
  height: fit-content;
  border-radius: 14px;
  box-shadow: 0 3px 1px -2px rgba(0, 0, 0, 0.2), 0 2px 2px 0 rgba(0, 0, 0, 0.14),
    0 1px 5px 0 rgba(0, 0, 0, 0.12);
  display: flex;
  flex-direction: column;
}

.messageText {
  margin-left: 1vw;
  margin-right: 1vw;
  margin-bottom: 1vh;
  margin-top: 1vh;
  word-wrap: break-word;
  max-width: 40vw;
}

.messageTimeSent {
  margin-right: 1vw;
  margin-left: 1vw;
  margin-bottom: 1vh;
  margin-top: 1vh;
  text-align: right;
}
