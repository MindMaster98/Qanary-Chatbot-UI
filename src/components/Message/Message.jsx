import React, { Component } from "react";
import { ClickableIcon } from "..";
import "./Message.css";
import account_icon_black from "../share/imgs/account_icon_black.svg";
import { textsHelper } from "../../helpers";

export default class Message extends Component {
  texts = textsHelper.getTexts();

  componentDidMount() {
    this.listenerId = textsHelper.addListener(() => {
      this.texts = textsHelper.getTexts();
      this.forceUpdate();
    });
  }

  componentWillUnmount() {
    textsHelper.removeListener(this.listenerId);
  }

  render() {
    return (
      <div
        tabIndex="-1"
        className={
          "messageContainer" +
          (!!this.props.messageObject?.isReply ? " reply" : "")
        }
      >
        <ClickableIcon
          alt="Avatar"
          icon={this.props.messageObject?.icon ?? account_icon_black}
          tabIndex={"-1"}
          style={{
            width: "100px",
            height: "100px",
            cursor: "initial",
          }}
        />
        <div
          tabIndex="0"
          className={
            "messageDataContainer"
          }
        >
          <div
            className={
              "messageText" +
              (!!this.props.messageObject?.isReply
                ? " white-text"
                : " black-text")
            }
            dangerouslySetInnerHTML={{
              __html: this.props.messageObject?.text ?? "",
            }}
          >
          </div>
          {!this.props.messageObject?.follow_up_needed &&
          !!this.props.messageObject?.isReply &&
          !!this.props.messageObject?.loadedSuccessfully ? (
            <div
              className={
                "messageText" +
                (!!this.props.messageObject?.isReply
                  ? " white-text"
                  : " black-text")
              }
            >
              {this.texts["message-info"]["source-of-data"]}
              <a
                href="https://experience.arcgis.com/experience/478220a4c454480e823b17327b2bf1d4/page/page_0/"
                rel="noreferrer"
                target="_blank"
              >
                {this.texts["message-info"]["robert-koch-institute"]}
              </a>{" "}
            </div>
          ) : null}
          <div className="messageTimeSent">
            {this.props.messageObject?.time}
          </div>
        </div>
      </div>
    );
  }
}
