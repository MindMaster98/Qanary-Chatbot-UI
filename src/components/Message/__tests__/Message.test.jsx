import renderer from "react-test-renderer";
import Message from "../Message";

// Snapshot tests

it("renders correctly with no props", () => {
  const tree = renderer.create(<Message />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly with valid reply message object, including disclaimer", () => {
  const tree = renderer.create(<Message messageObject={{
    isReply: true,
    loadedSuccessfully: true,
    followUpNeeded: false,
    time: "13:37",
  }} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly with valid user message object, without disclaimer", () => {
  const tree = renderer.create(<Message messageObject={{
    isReply: false,
    loadedSuccessfully: true,
    followUpNeeded: false,
    time: "13:37",
  }} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly with undefined message object properties", () => {
  const tree = renderer.create(<Message messageObject={{
    isReply: undefined,
    loadedSuccessfully: undefined,
    followUpNeeded: undefined,
    time: undefined,
    icon: undefined,
  }} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly with invalid icon as reply", () => {
  const tree = renderer.create(<Message messageObject={{
    isReply: true,
    loadedSuccessfully: true,
    followUpNeeded: false,
    time: "13:37",
    icon: "invalid",
  }} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly with invalid icon as user message", () => {
  const tree = renderer.create(<Message messageObject={{
    isReply: false,
    loadedSuccessfully: true,
    followUpNeeded: false,
    time: "13:37",
    icon: "invalid",
  }} />).toJSON();
  expect(tree).toMatchSnapshot();
});