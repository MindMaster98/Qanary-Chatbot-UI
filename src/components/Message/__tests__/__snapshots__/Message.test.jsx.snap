// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with invalid icon as reply 1`] = `
<div
  className="messageContainer reply"
  tabIndex="-1"
>
  <div
    className="clickableIcon "
    style={
      Object {
        "cursor": "initial",
        "height": "100px",
        "width": "100px",
      }
    }
    tabIndex="-1"
  >
    <img
      alt="Coronabot"
      src="invalid"
      tabIndex="-1"
    />
  </div>
  <div
    className="messageDataContainer"
    tabIndex="0"
  >
    <div
      className="messageText white-text"
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
    />
    <div
      className="messageText white-text"
    >
      Datenquelle: 
      <a
        href="https://experience.arcgis.com/experience/478220a4c454480e823b17327b2bf1d4/page/page_0/"
        rel="noreferrer"
        target="_blank"
      >
        Robert-Koch-Institut
      </a>
       
    </div>
    <div
      className="messageTimeSent"
    >
      13:37
    </div>
  </div>
</div>
`;

exports[`renders correctly with invalid icon as user message 1`] = `
<div
  className="messageContainer"
  tabIndex="-1"
>
  <div
    className="clickableIcon "
    style={
      Object {
        "cursor": "initial",
        "height": "100px",
        "width": "100px",
      }
    }
    tabIndex="-1"
  >
    <img
      alt="Nutzer"
      src="invalid"
      tabIndex="-1"
    />
  </div>
  <div
    className="messageDataContainer"
    tabIndex="0"
  >
    <div
      className="messageText black-text"
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
    />
    <div
      className="messageTimeSent"
    >
      13:37
    </div>
  </div>
</div>
`;

exports[`renders correctly with no props 1`] = `
<div
  className="messageContainer"
  tabIndex="-1"
>
  <div
    className="clickableIcon "
    style={
      Object {
        "cursor": "initial",
        "height": "100px",
        "width": "100px",
      }
    }
    tabIndex="-1"
  >
    <img
      alt="Nutzer"
      src="account_icon_black.svg"
      tabIndex="-1"
    />
  </div>
  <div
    className="messageDataContainer"
    tabIndex="0"
  >
    <div
      className="messageText black-text"
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
    />
    <div
      className="messageTimeSent"
    />
  </div>
</div>
`;

exports[`renders correctly with undefined message object properties 1`] = `
<div
  className="messageContainer"
  tabIndex="-1"
>
  <div
    className="clickableIcon "
    style={
      Object {
        "cursor": "initial",
        "height": "100px",
        "width": "100px",
      }
    }
    tabIndex="-1"
  >
    <img
      alt="Nutzer"
      src="account_icon_black.svg"
      tabIndex="-1"
    />
  </div>
  <div
    className="messageDataContainer"
    tabIndex="0"
  >
    <div
      className="messageText black-text"
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
    />
    <div
      className="messageTimeSent"
    />
  </div>
</div>
`;

exports[`renders correctly with valid reply message object, including disclaimer 1`] = `
<div
  className="messageContainer reply"
  tabIndex="-1"
>
  <div
    className="clickableIcon "
    style={
      Object {
        "cursor": "initial",
        "height": "100px",
        "width": "100px",
      }
    }
    tabIndex="-1"
  >
    <img
      alt="Coronabot"
      src="account_icon_black.svg"
      tabIndex="-1"
    />
  </div>
  <div
    className="messageDataContainer"
    tabIndex="0"
  >
    <div
      className="messageText white-text"
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
    />
    <div
      className="messageText white-text"
    >
      Datenquelle: 
      <a
        href="https://experience.arcgis.com/experience/478220a4c454480e823b17327b2bf1d4/page/page_0/"
        rel="noreferrer"
        target="_blank"
      >
        Robert-Koch-Institut
      </a>
       
    </div>
    <div
      className="messageTimeSent"
    >
      13:37
    </div>
  </div>
</div>
`;

exports[`renders correctly with valid user message object, without disclaimer 1`] = `
<div
  className="messageContainer"
  tabIndex="-1"
>
  <div
    className="clickableIcon "
    style={
      Object {
        "cursor": "initial",
        "height": "100px",
        "width": "100px",
      }
    }
    tabIndex="-1"
  >
    <img
      alt="Nutzer"
      src="account_icon_black.svg"
      tabIndex="-1"
    />
  </div>
  <div
    className="messageDataContainer"
    tabIndex="0"
  >
    <div
      className="messageText black-text"
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
    />
    <div
      className="messageTimeSent"
    >
      13:37
    </div>
  </div>
</div>
`;
